source/dynk.f90

```diff
@@ -321,14 +319,15 @@ subroutine dynk_parseFUN(inLine, iErr)
     dynk_ncData = dynk_ncData+1
 
     ! Open the file
-    inquire(unit=dynk_fileUnitFUN,opened=isOpen)
+    call f_requestUnit(dynk_cData(dynk_ncData),tmpUnit)
+    inquire(unit=tmpUnit,opened=isOpen)
```

Comment:
Nope. This is still useful to catch the error with a reasonable error message.

Reply:
So, `f_requestunit` doesn't check that the units it is providing is actually free?

Reply:
`f_requestUnit` doesn't check anything like that. It returns a unit from a list of available units. It will skip units that are opened though.

`f_open` has a catch all check for I/O errors, but that's not a substitute for an opened check.

Reply:
If it skips units that are opened, how can this then ever be triggered?

Reply:
Because a unit is only assigned to a file if it hasn't been assigned before. The open check is only performed when a unit is first assigned. The internal check is there to handle fixed units taken without a request, so for a different purpose than the one used in the line after. The second check can still in principle be triggered when the variable is reused, but it would indicate a bug.

Reply:
Right, because f_requestUnit will return the same unit again if called repeatedly with the same file name and the unit has not been freed (likely). So if the file wasn't closed the first time around it'll fail.

Fine.